python/cudf/cudf/tests/test_dataframe.py

```diff
@@ -4136,7 +4130,7 @@ def test_df_astype_string_to_other(as_dtype):
         data = ["2001-01-01", "2002-02-02", "2000-01-05", None]
         kwargs = {"format": "%Y-%m-%d"}
     elif as_dtype == "int32":
-        data = [1, 2, 3, None]
+        data = [1, 2, 3]
```

Comment:
Can we make the float explicitly `NaN` and datetime explicitly `NaT` to be more clear on the expected behavior? I.E. this behavior may implicitly change in Pandas 1.0+ with `pd.NA`.

Reply:
This has been modified, except DateTime for which there seems to be an issue: https://github.com/rapidsai/cudf/issues/5117